app/code/Magento/InventoryCatalog/Observer/ProcessSourceItemsObserver.php

```diff
@@ -106,14 +115,14 @@ public function execute(EventObserver $observer)
             $this->sourceItemsProcessor->process($productData['sku'], $assignedSources);
         } elseif (!empty($singleSourceData)) {
             /** @var StockItemInterface $stockItem */
-            $stockItem = $product->getExtensionAttributes()->getStockItem();
-            $qty = $singleSourceData['qty'] ?? (empty($stockItem) ? 0 : $stockItem->getQty());
-            $isInStock = $singleSourceData['is_in_stock'] ?? (empty($stockItem) ? 1 : (int)$stockItem->getIsInStock());
+            $outdatedStockItem = $product->getExtensionAttributes()->getStockItem();
```

Comment:
You can use `$legacyStockItem` instead of `$outdatedStockItem`

Reply:
@naydav, Outdated in this case mean not that "no longer supported" - it means that during request this data was changes and it is outdated :) 
So I agree that it can be rename to `$outdatedLegacyStockItem`.